x/emissions/migrations/v5/migrate.go

```diff
@@ -0,0 +1,198 @@
+	emissionsKeeper keeper.Keeper,
+) error {
+	topicStore := prefix.NewStore(store, emissionstypes.TopicsKey)
+	iterator := topicStore.Iterator(nil, nil)
+
+	// iterate over the topic store by manually checking prefixes
+	// to make sure we absolutely do not miss any topics
+	// (as opposed to using collections.go API)
+	topicsToChange := make(map[string]emissionstypes.Topic, 0)
+	for ; iterator.Valid(); iterator.Next() {
+		ctx.Logger().Info("MIGRATION V5: Updating topic", iterator.Key())
+		iterator.Key()
+		var oldMsg oldV2Types.Topic
+		err := proto.Unmarshal(iterator.Value(), &oldMsg)
+		if err != nil {
+			return errorsmod.Wrapf(err, "failed to unmarshal old topic")
+		}
+		// now get the topic from the collections.go
+		// API, to check if the fields in the topic exist
+		topic, err := emissionsKeeper.GetTopic(ctx, oldMsg.Id)
+		if err != nil {
+			return errorsmod.Wrapf(err, "failed to get topic")
+		}
+		newTopic := copyTopic(topic)
+		// fix egregious values
+		newTopic.InitialRegret = alloraMath.MustNewDecFromString("0")
+		// was wrongly set to true for existing topics
+		newTopic.AllowNegative = false
+		topicsToChange[string(iterator.Key())] = newTopic
+	}
+	_ = iterator.Close()
+	for key, value := range topicsToChange {
+		topicStore.Set([]byte(key), cdc.MustMarshal(&value))
+	}
+
+	return nil
+}
+
+// Deletes all keys in the store with the given keyPrefix `maxPageSize` keys at a time
+func safelyClearWholeMap(ctx sdk.Context, store storetypes.KVStore, keyPrefix []byte) {
+	s := prefix.NewStore(store, keyPrefix)
+
+	// Loop until all keys are deleted.
+	// Unbounded not best practice but we are sure that the number of keys will be limited
+	// and not deleting all keys means "poison" will remain in the store.
+	for {
+		// Gather keys to eventually delete
+		iterator := s.Iterator(nil, nil)
+		keysToDelete := make([][]byte, 0)
+		count := uint64(0)
+		for ; iterator.Valid(); iterator.Next() {
+			ctx.Logger().Info("MIGRATION V5: DELETING keys in store with prefix", "prefix", keyPrefix, "page", count)
+			if count >= maxPageSize {
+				break
+			}
+
+			keysToDelete = append(keysToDelete, iterator.Key())
+			count++
+		}
+		iterator.Close()
+
+		// If no keys to delete, break => Exit whole function
+		if len(keysToDelete) == 0 {
+			break
+		}
+
+		// Delete the keys
+		for _, key := range keysToDelete {
+			s.Delete(key)
+		}
+	}
+}
+
+// Clear out poison NaN values on different inferences, scores etc
+func ResetMapsWithNonNumericValues(ctx sdk.Context, store storetypes.KVStore, cdc codec.BinaryCodec) {
+	ctx.Logger().Info("MIGRATION V4: RESETTING infererScoresByBlock MAP")
+	safelyClearWholeMap(ctx, store, emissionstypes.InferenceScoresKey)
+	ctx.Logger().Info("MIGRATION V4: RESETTING forecasterScoresByBlock MAP")
+	safelyClearWholeMap(ctx, store, emissionstypes.ForecastScoresKey)
+	ctx.Logger().Info("MIGRATION V4: RESETTING reputerScoresByBlock MAP")
+	safelyClearWholeMap(ctx, store, emissionstypes.ReputerScoresKey)
```

Comment:
do we really need to clear out the score maps? regrets aren't scores?

Reply:
Yes I don't think we _need_ to clear scores or ema_scores afaik 

Same for coefficients, quantiles, reward fractions I think